tasks/eva_2326/src/main/groovy/eva2326/update_contig_name.groovy

```diff
@@ -0,0 +1,593 @@
+
+                    annotationsToBeUpdated.add(orgVariant.getId())
+                    idProcessed.add(newId)
+
+                    continue
+                }
+
+                logger.info("can't merge as sid fid common pair has more than 1 entry in file")
+                storeVariantsThatCantBeProcessed(variantsWithIssuesFilePath, orgVariant.getId(), variantInDB.getId(), "Can't merge as sid fid common pair has more than 1 entry in file")
+
+            }
+
+            // delete existing variants
+            if (!documentsToDeleteIdList.isEmpty()) {
+                mongoTemplate.remove(Query.query(Criteria.where("_id").in(documentsToDeleteIdList)), VARIANTS_COLLECTION)
+            }
+
+            // insert the updated variants
+            if (!documentsToInsertList.isEmpty()) {
+                List<Document> documentsToInsert = documentsToInsertList.stream()
+                        .map(variantDocument -> mongoTemplate.getConverter().convertToMongoType(variantDocument))
+                        .collect(Collectors.toList())
+                mongoTemplate.getCollection(VARIANTS_COLLECTION).insertMany(documentsToInsert)
+            }
+
+            // perform the batched update
+            if (!bulkUpdateOperations.isEmpty()) {
+                mongoTemplate.getCollection(VARIANTS_COLLECTION).bulkWrite(bulkUpdateOperations)
+            }
+
+            if (!annotationsToBeUpdated.isEmpty()) {
+                orgVariantNewIdMap = orgVariantNewIdMap.findAll { k, v -> annotationsToBeUpdated.contains(k) }
+                remediateAnnotations(orgVariantNewIdMap)
+            }
+
+
+            variantDocumentList = processInNextLoop
+
+        }
+    }
+
+    List<VariantDocument> getVariantsInDB(List<String> idList) {
+        Query idQuery = new Query(where("_id").in(idList))
+        List<VariantDocument> variantInDBList = mongoTemplate.find(idQuery, VariantDocument.class, VARIANTS_COLLECTION)
+
+        return variantInDBList
+    }
+
+    void storeVariantsThatCantBeProcessed(String variantsWithIssuesFilePath, String variantId, String newVariantId, String reason) {
+        try (BufferedWriter variantsWithIssueFile = new BufferedWriter(new FileWriter(variantsWithIssuesFilePath, true))) {
+            variantsWithIssueFile.write(variantId + "," + newVariantId + "," + reason + "\n")
+        } catch (IOException e) {
+            logger.error("error storing variant id that can't be processed in the file:  {}", variantId)
+        }
+    }
+
+    VariantDocument remediateCaseNoIdCollision(VariantDocument orgVariant, String newId, String updatedChromosome) {
+        Map<String, Set<String>> updatedHgvs = getUpdatedHgvs(orgVariant, updatedChromosome)
+        Set<VariantSourceEntryMongo> updatedSourceEntries = orgVariant.getVariantSources().stream()
+                .peek(vse -> {
+                    if (vse.getAttrs() != null) {
+                        vse.getAttrs().append("CHR", updatedChromosome)
```

Comment:
Probably some discussion I forgot about, but would it make more sense to store the original contig name in the `files` attributes? Since the updated contig is already in the `chr` of the variant document.

Additionally, does appending to `attrs` in this way mean that it doesn't get included in the `src` attribute via [this method](https://github.com/EBIvariation/eva-pipeline/blob/master/src/main/java/uk/ac/ebi/eva/commons/models/mongo/entity/subdocuments/VariantSourceEntryMongo.java#L164)?  I'm not sure we'd want it to be included since it wasn't part of the original VCF, but just wanted to confirm that's what we want.

Reply:
removed